processout-sdk/src/main/java/com/processout/processout_sdk/ProcessOut.java

```diff
@@ -233,20 +242,108 @@ public void error() {
                     }
                 });
                 break;
+            case REDIRECT:
             case URL:
+                // Create external webview
                 WebView theWebPage = new WebView(this.context);
                 theWebPage.getSettings().setJavaScriptEnabled(true);
                 theWebPage.getSettings().setPluginState(WebSettings.PluginState.ON);
                 Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(cA.getValue()));
                 this.context.startActivity(browserIntent);
                 break;
+            case FINGERPRINT:
+                // Build the fingerprint hidden webview
+                final WebView fingerPrintWebView = new WebView(this.context);
+                fingerPrintWebView.getSettings().setJavaScriptEnabled(true); // enable javascript
+                fingerPrintWebView.setVisibility(View.INVISIBLE); // Hide the Webview
+
+                // Defining fallback request in case the fingerprint times out or is unavailable
+                MiscGatewayRequest fallbackGwayRequest = new MiscGatewayRequest("{\"threeDS2FingerprintTimeout\":true}");
```

Comment:
You also want to set its URL to the Fingerprint URL, as well as the headers to `Content-Type: application/json`